MekHQ/src/mekhq/campaign/autoResolve/helper/RandomUtils.java

```diff
@@ -0,0 +1,187 @@
+package mekhq.campaign.autoResolve.helper;
+
+import org.apache.commons.lang3.stream.Streams;
+
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class RandomUtils {
```

Comment:
We could do with some JavaDocs for this class, and/or methods. It might also be worth taking a look at `ObjectUtility` to see whether there is value in merging the two classes. I would have mentioned this back when you first suggested a random workhorse method, but I only discovered it existed earlier today.